src/shared/protocol.test.ts

```diff
@@ -62,6 +62,182 @@ describe("protocol tests", () => {
     await transport.close();
     expect(oncloseMock).toHaveBeenCalled();
   });
+
+  test("should reset timeout when progress notification is received", async () => {
+    jest.useFakeTimers();
+    
+    await protocol.connect(transport);
+    const request = { method: "example", params: {} };
+    const mockSchema: ZodType<{ result: string }> = z.object({
+      result: z.string(),
+    });
+
+    const onProgressMock = jest.fn();
+    const requestPromise = protocol.request(request, mockSchema, {
+      timeout: 1000,  // Increased timeout for more reliable testing
+      resetTimeoutOnProgress: true,
+      onprogress: onProgressMock,
+    });
+
+    // Advance time close to timeout
+    jest.advanceTimersByTime(800);
+
+    // Send progress notification
+    if (transport.onmessage) {
+      transport.onmessage({
+        jsonrpc: "2.0",
+        method: "notifications/progress",
+        params: {
+          progressToken: 0,
+          progress: 50,
+          total: 100,
+        },
+      });
+    }
+
+    // Run all pending promises to ensure progress handler is called
+    await Promise.resolve();
+
+    // Verify progress handler was called
+    expect(onProgressMock).toHaveBeenCalledWith({
+      progress: 50,
+      total: 100,
+    });
+
```

Comment:
Presumably we need to advance the timers again here? Otherwise this doesn't seem to really be testing that the timeout got extended.